d2l-image.js

```diff
@@ -16,66 +32,59 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-image">
 				border-radius: var(--d2l-image-border-radius);
 				object-fit: var(--d2l-image-object-fit);
 			}
-		</style>
+		`;
+	}
 
-		<iron-ajax id="imageRequest" url="[[imageUrl]]" headers="[[_headers]]" on-iron-ajax-response="_onImageResponse" on-iron-ajax-error="_onImageResponse" handle-as="blob">
-		</iron-ajax>
+	constructor() {
+		super();
+		window.d2lfetch.use({ name: 'auth', fn: auth });
+	}
 
-		<img id="image" alt$="[[alternateText]]">
-	</template>
+	render() {
+		if (!this._imageUrl) {
+			return nothing;
+		}
 
-</dom-module>`;
+		return html`
+			<img alt="${this.altText}" src="${ifDefined(this._imageUrl)}">
+		`;
+	}
 
-document.head.appendChild($_documentContainer.content);
+	updated(changedProperties) {
+		if (changedProperties.has('imageUrl')) {
+			this._loadImage();
+		}
+	}
 
-Polymer({
-	is: 'd2l-image',
+	async _loadImage() {
+		if (!this.imageUrl) {
+			return;
+		}
 
-	properties: {
-		alternateText: {
-			type: String,
-			reflectToAttribute: true,
-		},
-		imageUrl: {
-			type: String,
-			value: null
-		},
-		token: String,
-		_headers: String
-	},
+		let response;
 
-	observers: [
-		'_onImageUrlChange( imageUrl, token )'
-	],
+		try {
+			response = await window.d2lfetch.fetch(this.imageUrl);
 
-	_onImageUrlChange: function(imageUrl, token) {
-		if (imageUrl && token) {
-			this._headers = {
-				Authorization: 'Bearer ' + token
-			};
-			this.$.imageRequest.generateRequest();
-		} else if (imageUrl) {
-			this.$.image.src = imageUrl;
-		}
-	},
+			const blob = await response.blob();
+			this._imageUrl = URL.createObjectURL(blob);
 
-	_onImageResponse: function(response) {
-		if (response.detail.status === 200) {
-			if (this.$.image.src) {
-				URL.revokeObjectURL(this.$.image.src);
-			}
-			this.$.image.src = URL.createObjectURL(response.detail.response);
 			this.dispatchEvent(new CustomEvent('d2l-image-loaded', {
 				bubbles: true,
 				composed: true,
```

Comment:
There's only [1 usage](https://search.d2l.dev/xref/Brightspace/manager-view-fra/src/app/mr-image.js?r=83fac7ca#31) of this event and [3 usages](https://search.d2l.dev/search?project=Brightspace&project=BrightspaceHypermediaComponents&project=BrightspaceUI&project=BrightspaceUILabs&project=lms&full=%22d2l-image-failed-to-load%22&defs=&refs=&path=&hist=&type=&xrd=&nn=5&si=full&si=full) of the failed event, and they all wire up directly to the `<d2l-image>` element. So I think we could change this to `bubbles: false` and `composed: false` to reduce the footprint of the event.